src/system/SystemLayer.am

```diff
@@ -28,6 +28,7 @@ CHIP_BUILD_SYSTEM_LAYER_SOURCE_FILES                  = \
     @top_builddir@/src/system/SystemLayer.cpp           \
     @top_builddir@/src/system/SystemMutex.cpp           \
     @top_builddir@/src/system/SystemObject.cpp          \
+    @top_builddir@/src/system/SystemWakeEvent.cpp       \
```

Comment:
please separate adding this functionality into a separate PR that includes tests for the module

Reply:
ok, can do it

Reply:
see #1594